File: Third Semester - Advanced Programming Methods/Lab10/factory/TaskContainerFactory.cs
using lab10.model;

namespace lab10.factory;

public class TaskContainerFactory: IFactory
{
    private static TaskContainerFactory _instance;
    
    private TaskContainerFactory() { }
    
    public static TaskContainerFactory Instance => _instance ??= new TaskContainerFactory();
    
    public IContainer createContainer(Strategy startegy)
    {
        switch (startegy)
        {
            case Strategy.LIFO: return new StackContainer();
            case Strategy.FIFO: return new QueueContainer();
            default: throw new ArgumentException("Invalid strategy");
        }
    }
}

File: Third Semester - Advanced Programming Methods/Lab2/factory/TaskContainerFactory.java
package factory;

import model.Container;
import model.QueueContainer;
import model.StackContainer;

public class TaskContainerFactory implements Factory{
    private static TaskContainerFactory instance = null ;

    private TaskContainerFactory() {}

    public static TaskContainerFactory getInstance() {
        if(instance == null) {
            instance = new TaskContainerFactory();
        }
        return instance;
    }

    @Override
    public Container createContainer(Strategy strategy) {
        return switch (strategy){
            case FIFO -> new QueueContainer();
            case LIFO -> new StackContainer();
            default -> throw new IllegalArgumentException("Invalid strategy");
        };
    }
}
